### ISSUE TITLE
🪲 Saving turtle drawings is not possible and results in a hidden (console log) error

### ISSUE THREAD

**Issue by mani1986** [hedyorg/hedy#5123]
**Describe the bug**
When I draw a turtle from the example code from level 5 / turtle then it appears that nothing happens. When I look at the console log I see that the request to `/generate_machine_files` fails with a 500 error.

**Paste the Hedy code & level**
I am on level 5 / turtle.
The code:
```
figure is ask 'Do you want a square or a triangle?'
print 'Drawing Figures'
if figure is triangle angle is 120
else angle is 90
turn angle
forward 25
turn angle
forward 25
turn angle
forward 25
turn angle
forward 25

```

**Add a screenshot (optional)**
<img width="1714" alt="Screenshot 2024-02-09 at 15 43 58" src="https://github.com/hedyorg/hedy/assets/739215/9181db10-943c-47b2-a65f-0e342f50503b">

**What machine and browser you were using (optional)**
- What computer you are using Mac
- What browser you were using Chrome

**Comment by jpelay** [comment]
After some detective work I found out that the actual error returned by the application is this:

```
File "/app/.heroku/python/lib/python3.9/site-packages/turtlethread/stitches.py", line 166, in _get_stitch_commands
    for pos1, pos2 in itertools.pairwise(self._positions):
AttributeError: module 'itertools' has no attribute 'pairwise'
```

The problem being that the library we use to download the turtle files is using the `pairwise` function that only becomes available in the `itertools` module after Python 3.10 and we are currently using Python 3.9 on our Heroku applications, what do you think @Felienne?

**Comment by Felienne** [comment]
> After some detective work I found out that the actual error returned by the application is this:
> 
> ```
> File "/app/.heroku/python/lib/python3.9/site-packages/turtlethread/stitches.py", line 166, in _get_stitch_commands
>     for pos1, pos2 in itertools.pairwise(self._positions):
> AttributeError: module 'itertools' has no attribute 'pairwise'
> ```
> 
> The problem being that the library we use to download the turtle files is using the `pairwise` function that only becomes available in the `itertools` module after Python 3.10 and we are currently using Python 3.9 on our Heroku applications, what do you think @Felienne?

Ah interesting! They must have updated this at one point, because it used to work when I created it. Shall we try an update to Python 3.10 (we must do that at ont point, anyway)

**Comment by Felienne** [comment]
I am trying in this PR, let's see what happens: #5134

**Comment by Felienne** [comment]
Also, I have been wanting to improve the turtle downloading for a long time, but I never got around to it, so I am dumping some issues here:

**We should not rerun!**
What we do now, it rerunning the whole program to create the dst-file and then save it, but that it not great for several reasons (programs with an `ask` block and never creates a file and if you use `random` you can get a different picture). What we should do instead is have a toggle in the UI, rather then a button, and when the toggle is toggled, create the dst right away.

**Downloading multiple files does not work well**
I tried to download 2 files so people could also run the files on other devices like a penplotter. But PNG does not work with a plotter so we should convert to SVG or something

**Comment by Felienne** [comment]
Thanks for letting us know @mani1986!! This is fixed by updating Python to 3.12 (now running on alpha)
